website/versioned_docs/version-5.x/server-side-rendering-overview.md

```diff
@@ -0,0 +1,294 @@
+
+## Implementation Overview
+
+The ultimate goal of server-side rendering is to generate an HTTP response that the browser will display to the user while javascript is hydrating. Most microfrontend server-side rendering implementations, including single-spa's recommended approach, do this with the following steps:
+
+1. Layout - Identify which microfrontends to render for the incoming HTTP request, and where within the HTML they will be placed. This is usually route based.
+2. Fetch - Begin rendering the HTML for each microfrontend to a stream.
+3. Headers - Retrieve HTTP response headers from each microfrontend. Merge them together and send the result as the HTTP response headers to the browser.
+4. Body - Send the HTTP response body to the browser, which is an HTML document consisting of static and dynamic parts. This involves waiting for each microfrontend's stream to end before proceeding to the next portion of HTML.
+5. Hydrate - Within the browser, download all javascript needed to hydrate the HTML.
+
+## 1. Layout
+
+To define an HTML template that lays out your page, first choose a "microfrontend layout middleware":
+
+1. [single-spa-layout](/docs/layout-overview): The official layout engine for single-spa.
+2. [Tailor](https://github.com/zalando/tailor): A popular, battle tested layout engine that predates single-spa-layout and is not officially affiliated with single-spa.
+3. [TailorX](https://github.com/StyleT/tailorx): An actively maintained fork of Tailor that is used by [Namecheap](https://www.namecheap.com/) in their single-spa website. The single-spa core team collaborated with the creators of TailorX when authoring single-spa-layout, taking some inspiration from it.
+
+We generally recommend single-spa-layout, although choosing one of the other options might make sense for your situation, since single-spa-layout is newer and less battle tested than Tailor/TailorX.
+
+With single-spa-layout, you define a single template that handles all routes. [Full documentation](/docs/layout-definition).
+
+```html
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="UTF-8" />
+    <meta name="viewport" content="width=device-width, initial-scale=1.0" />
+    <meta http-equiv="X-UA-Compatible" content="ie=edge" />
+    <title>Isomorphic Microfrontends</title>
+    <meta
+      name="importmap-type"
+      content="systemjs-importmap"
+      server-cookie
+      server-only
+    />
+    <script src="https://cdn.jsdelivr.net/npm/import-map-overrides@2.0.0/dist/import-map-overrides.js"></script>
+    <script src="https://cdn.jsdelivr.net/npm/systemjs@6.6.1/dist/system.min.js"></script>
+    <script src="https://cdn.jsdelivr.net/npm/systemjs@6.6.1/dist/extras/amd.min.js"></script>
+    <script src="https://cdn.jsdelivr.net/npm/systemjs@6.6.1/dist/extras/named-exports.min.js"></script>
+  </head>
+  <body>
+    <template id="single-spa-layout">
```

Comment:
The [Layout Definition](https://single-spa.js.org/docs/layout-definition/#html-layouts) docs put the `template` element in the `head`. I vaguely recall that that is to prevent it from being rendered, but with SSR that doesn't matter because its being pre-rendered (if I'm not mistaken... which I'm not confident I am). Is that worth noting somewhere? Might be better to put over in that page instead though... 🤔

Reply:
[Template elements are rendered in IE](https://stackoverflow.com/a/45680787/2554793), but of course that's on the client. I'm just carrying over what I remember from that example code since it might be a confusing point.